dbms/src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -0,0 +1,442 @@
+#include <Dictionaries/getDictionaryConfigurationFromAST.h>
+
+#include <Poco/DOM/AutoPtr.h>
+#include <Poco/DOM/Document.h>
+#include <Poco/DOM/Element.h>
+#include <Poco/DOM/Text.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <Poco/Util/XMLConfiguration.h>
+#include <IO/WriteHelpers.h>
+#include <Parsers/queryToString.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTFunction.h>
+#include <Core/Names.h>
+#include <Parsers/ASTFunctionWithKeyValueArguments.h>
+#include <Parsers/ASTDictionaryAttributeDeclaration.h>
+#include <Dictionaries/DictionaryFactory.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int INCORRECT_DICTIONARY_DEFINITION;
+}
+
+/// There are a lot of code, but it's very simple and straightforward
+/// We just convert
+namespace
+{
+
+String unescapeString(const String & string)
+{
+    if (!string.empty() && string.front() == '\'' && string.back() == '\'')
```

Comment:
Is it OK for string with one symbol ```'```?

Reply:
Actually, it's impossible here. This function is used in a single place, where we converting attribute type to string with `applyVisitor(FieldVisitorToString(), literal->value)`. I didn't find any other way to convert field value to string without quotes :(.